catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+      case ZONE:
+        LOGGER.trace("Saving this part");
+        mgrsBuilder.append(mgrsPart);
+        break;
+      case HUNDRED_KM:
+      case NUMERIC:
+      case NONE:
+        LOGGER.trace("Ignoring this part, since 100KM is only valid after a ZONE");
+        break;
+    }
+  }
+
+  private static void handleNumericWithPreviousType(
+      String mgrsPart,
+      MgrsPartType previousType,
+      StringBuilder mgrsBuilder,
+      List<String> mgrsStrings) {
+    LOGGER.trace("Found a NUMERIC [{}] which occurs after a {}", mgrsPart, previousType);
+    switch (previousType) {
+      case ZONE:
+      case HUNDRED_KM:
+      case NUMERIC:
+        String processed = processNumericType(mgrsPart);
+        LOGGER.trace("NUMERIC [{}] processed = [{}]", mgrsPart, processed);
+        LOGGER.trace("Storing final result coordinate");
+        mgrsStrings.add(mgrsBuilder.toString() + processed);
+        break;
+      case NONE:
+        LOGGER.trace(
+            "Ignoring this part, since NUMERIC is only valid in a ZONE but no ZONE was provided");
+        break;
+    }
+  }
+
+  /**
+   * Given an MGRS {@link MgrsPartType#NUMERIC} part format the string to ensure an accurate
+   * conversion, using whitespace as a hint as to the user's intentions.
+   *
+   * <p>The usng4j library demands a minimum of 6 symbols if the easting / northing data is not to
+   * be ignored. That is, the easting component must be fully specified for the conversion to work.
+   * To accomodate this, zeros are prepended onto the easting to whatever appropriate degree.
+   *
+   * <p>For example, an input of <span>4Q KJ 1 2</span> will become <span>4QKJ000012</span> so the
+   * easting / northing is not ignored.
+   *
+   * <p>Refer to the unit tests for more comprehensive examples.
+   *
+   * <p>TODO DDF-4243 - This should be simplified if usng4j is improved
+   *
+   * @see #assemble(String, String)
+   * @param mgrsPart an MGRS substring that contains BOTH the easting and northing.
+   * @return a modified easting/northing string with zeros backfilled to ensure accurate conversion.
+   */
+  private static String processNumericType(String mgrsPart) {
+    String[] numericParts = mgrsPart.split("[ ]+");
+    if (numericParts.length >= 3 || numericParts.length == 0) {
+      throw new IllegalArgumentException(
+          "The mgrs part provided is not a valid NUMERIC type: " + mgrsPart);
+    }
+    if (numericParts.length == 2) {
+      return assemble(numericParts[0], numericParts[1]);
+    }
+    String eastingAndNorthing = numericParts[0];
+    if (eastingAndNorthing.length() % 2 > 0) {
+      eastingAndNorthing = eastingAndNorthing.substring(0, eastingAndNorthing.length() - 1);
+    }
+    int halfway = eastingAndNorthing.length() / 2;
+    String easting = eastingAndNorthing.substring(0, halfway);
+    String northing = eastingAndNorthing.substring(halfway, eastingAndNorthing.length());
+    return assemble(easting, northing);
+  }
+
```

Comment:
✏️ `easting` and `northing` sound like verbs. What does these represent?

Reply:
> The terms easting and northing are geographic Cartesian coordinates for a point. Easting refers to the eastward-measured distance (or the x-coordinate), while northing refers to the northward-measured distance (or the y-coordinate).

https://en.m.wikipedia.org/wiki/Easting_and_northing